libc/stdlib_bsearch.c

```diff
@@ -0,0 +1,320 @@
+/*
+ * Phoenix-RTOS
+ *
+ *    POSIX.1-2017 standard library functions tests
+ *    HEADER:
+ *    - stdlib.h
+ *    TESTED:
+ *    - bsearch()
+ *
+ * Copyright 2023 Phoenix Systems
+ * Author: Adam Debek
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <string.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <unity_fixture.h>
+
+
+struct node {
+	char *string;
+	int length;
+};
+
+/* Special node with reversed attributes */
+struct node2 {
+	int length;
+	char *string;
+};
+
+
+/* table sorted alphabetically */
+static char *names[] = { "adam",
+	"bartek",
+	"cezary",
+	"damian",
+	"edward",
+	"fryderyk",
+	"gracjan",
+	"henryk",
+	"iwo",
+	"jan" };
+
+static char *namesRepeat[] = { "adam",
+	"adam",
+	"adam",
+	"damian",
+	"edward",
+	"fryderyk",
+	"fryderyk",
+	"henryk",
+	"henryk",
+	"jan" };
+
+/* table sorted alphabetically */
+static char *names_nsort[] = { "henryk",
+	"bartek",
+	"cezary",
+	"damian",
+	"edward",
+	"fryderyk",
+	"gracjan",
+	"adam",
+	"iwo",
+	"jan" };
+
+#define TABSIZE sizeof(names) / sizeof(names[0])
+
+static struct node test_table[TABSIZE];
+
+static void test_prepare(char *names_arg[])
+{
+	int i;
+
+	for (i = 0; i < TABSIZE; i++) {
+		test_table[i].string = (char *)malloc(strlen(names_arg[i]) + 1);
+
+		TEST_ASSERT_NOT_NULL(test_table[i].string);
+
+		strcpy(test_table[i].string, names_arg[i]);
+		test_table[i].length = strlen(names_arg[i]);
+	}
+}
+
+
+static void test_clean(char *names_arg[])
+{
+	int i;
+
+	memset(test_table, 0, sizeof(struct node) * TABSIZE);
+
+	for (i = 0; i < TABSIZE; i++) {
+		free(test_table[i].string);
+	}
+}
+
+
+/* bsearch can't change content of table, but may reorder elements */
+static int test_checkTab(char *names_arg[])
+{
+	int i, j, found;
+	for (i = 0; i < TABSIZE; i++) {
+		found = 0;
+		for (j = 0; j < TABSIZE; j++) {
+			if (strcmp(names_arg[i], test_table[j].string) == 0 && test_table[i].length == test_table[j].length) {
+				found = 1;
+				break;
+			}
+		}
+		if (found == 0) {
+			return found;
+		}
+	}
+
+	return found;
+}
+
+
+/* compare function used by bsearch */
+static int test_nodeCmp(const void *node1, const void *node2)
+{
+	return strcmp(((struct node *)node1)->string, ((struct node *)node2)->string);
+}
+
+
+/* compare function used by bsearch in special_node testcase */
+static int test_nodeCmp2(const void *node1, const void *node2)
```

Comment:
Please use more distinctive names.

```suggestion
static int test_nodeCmpStr(const void *node1, const void *node2)
{
	return strcmp(((struct node *)node1)->string, ((struct node *)node2)->string);
}


/* compare function used by bsearch in special_node testcase */
static int test_nodeCmpLen(const void *node1, const void *node2)
```